Library/Homebrew/livecheck/strategy/extract_plist.rb

```diff
@@ -82,25 +82,37 @@ def self.versions_from_items(items, regex = nil, &block)
         # versions from `plist` files.
         #
         # @param cask [Cask::Cask] the cask to check for version information
+        # @param url [String, nil] an alternative URL to check for version
+        #   information
         # @param regex [Regexp, nil] a regex for use in a strategy block
         # @return [Hash]
         sig {
           params(
             cask:    Cask::Cask,
+            url:     T.nilable(String),
             regex:   T.nilable(Regexp),
             _unused: T.nilable(T::Hash[Symbol, T.untyped]),
             block:   T.untyped,
           ).returns(T::Hash[Symbol, T.untyped])
         }
-        def self.find_versions(cask:, regex: nil, **_unused, &block)
+        def self.find_versions(cask:, url: nil, regex: nil, **_unused, &block)
           if regex.present? && block.blank?
             raise ArgumentError, "#{T.must(name).demodulize} only supports a regex when using a `strategy` block"
           end
           raise ArgumentError, "The #{T.must(name).demodulize} strategy only supports casks." unless T.unsafe(cask)
 
-          match_data = { matches: {}, regex: regex }
+          match_data = { matches: {}, regex: regex, url: url }
+
+          unversioned_cask_checker = if url.present? && url != cask.url.to_s
+            # Create a copy of the `cask` where the `url` is replaced with
+            # the `url` argument
+            cask_copy = Cask::CaskLoader.load(cask.full_name)
+            cask_copy.instance_variable_get(:@dsl).instance_variable_set(:@url, URL::DSL.new(url))
```

Comment:
@MikeMcQuaid Any objections to using `#instance_variable_set` here to replace the `@url` instance variable in a copy of the cask? As mentioned above, something like `cask_copy { url }` produces a `Cask::CaskInvalidError`, as `Cask::DSL#set_unique_stanza` prevents `@url` from being set more than once.

Reply:
Thanks @samford

Reply:
> If it matters, it's basically just replicating [what happens in Cask::DSL#set_unique_stanza](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/dsl.rb#L112-L124) but we're outside of the Cask::DSL object, so we also have to use instance_variable_get(:@dsl).

Yeh, I think it's different when it's happening within the class itself.

> Past that, I'm not sure how we would approach this within Cask, other than removing the "stanza may only appear once" restrictions (i.e., bringing it in line with the Formula DSL behavior). Was there anything in particular you were thinking of?

some options:
- allow optionally disabling the "stanza may only appear once" in this case (but not in the cask itself)
- add a new internal API for this case

Reply:
@samford Yeh, this would work for me!

Reply:
@samford maybe `allow_reassignment: false` by default or something?